### PR TITLE
task.h : close loop

### PR DESCRIPTION
**Pull request by ujsxn**
1.should use the uv_loop_close instead of uv_loop_delete.

### CODE REVIEW

File: test/task.h [hunk 1]
```diff
@@ -109,7 +109,7 @@ typedef enum {
 #define MAKE_VALGRIND_HAPPY()           \
   do {                                  \
     close_loop(uv_default_loop());      \
-    uv_loop_delete(uv_default_loop());  \
+    uv_loop_close(uv_default_loop());  \
```
Comment (saghul) [inline test/task.h]: ASSERT that the return value is 0.

File: test/task.h [hunk 2]
```diff
@@ -227,7 +227,6 @@ static void close_walk_cb(uv_handle_t* handle, void* arg) {
 
 UNUSED static void close_loop(uv_loop_t* loop) {
   uv_walk(loop, close_walk_cb, NULL);
-  uv_run(loop, UV_RUN_DEFAULT);
```
Comment (saghul) [inline test/task.h]: This is incorrect. The loop is stopped, so the close handles will not be called unless we run it again, so me must do it.

File: test/task.h [hunk 3]
```diff
@@ -109,7 +109,7 @@ typedef enum {
 #define MAKE_VALGRIND_HAPPY()           \
   do {                                  \
     close_loop(uv_default_loop());      \
-    uv_loop_delete(uv_default_loop());  \
+    ASSERT(uv_loop_close(uv_default_loop())==0);  \
```
Comment (saghul) [inline test/task.h]: leave spaces between ==

File: test/task.h [hunk 4]
```diff
@@ -227,7 +227,7 @@ static void close_walk_cb(uv_handle_t* handle, void* arg) {
 
 UNUSED static void close_loop(uv_loop_t* loop) {
   uv_walk(loop, close_walk_cb, NULL);
-  uv_run(loop, UV_RUN_DEFAULT);
+  uv_run(loop,UV_RUN_DEFAULT);
```
Comment (saghul) [inline test/task.h]: space after comma

**Comment by saghul** [comment]
Left a couple more comments. Please leave a note after you address them, GitHub won't send a notification otherwise.

**Comment by saghul** [comment]
CI: https://ci.nodejs.org/job/libuv+any-pr+multi/246/

**Comment by ujsxn** [comment]
Done.
as I never being into a project like this, thanks for your patient.

**Comment by ujsxn** [comment]
the last url you showed is:
"
Access Denied

ujsxn没有Overall/Read权限
"

I don's know what this mean, and whether it works.

**Comment by ujsxn** [comment]
And I found this:
1. if (handle == (uv_handle_t*)&timer)
2. ASSERT(handle == &timer);
as you indicate the coding style , maybe these should be CKECK_HANDLE

**Comment by saghul** [comment]
Thanks! Landed in b6650df. A few observations for next time:

* Follow the guidelines in CONTRIBUTING.md with regards to patch description
* Squash the commits into one once it's all ready
* Make your changes match the surrounding style

Feel free to ask any questions if you struggle with any of the above.

**Pull request closed without merging**